Feat/landing hero section

transcendence12 opened the pull request:
Create Hero Section

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HeroBottom/HeroBottom.module.scss:
```diff
@@ -0,0 +1,227 @@
+  margin-top: 40px;
+}
+
+@include mobile {
```
**suggestion**: Consider using a mixin for the repeated mobile styles to reduce redundancy.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HeroBottom/HeroBottom.tsx:
```diff
@@ -0,0 +1,309 @@
+  const tagsAnimate = tagKeys.map((key) => t(`animate_${key}`))
+
+  // Create rows based on device type
+  const getRows = () => {
```
**praise**: Great use of a function to dynamically create rows based on device type, enhancing responsiveness.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HeroBottom/HeroBottom.tsx:
```diff
@@ -0,0 +1,309 @@
+    tagMapping[animatedText] = staticText
+  })
+
+  const handleTagClick = (tag: string) => {
```
**suggestion**: Add error handling for the `handleTagClick` function to manage unexpected inputs.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HeroBottom/HeroBottom.tsx:
```diff
@@ -0,0 +1,309 @@
+  const mockText =
+    'Szukam rozwiązania w postaci prywatnego asystenta AI do wsparcia mojej pracy. Potrzebuję systemu, który:\nAutomatycznie zarządza kalendarzem i spotkaniami\nPrzetwarza i kategoryzuje emaile\nPrzygotowuje wstępne odpowiedzi na wiadomości\nPrzypomina o ważnych zadaniach i deadlinach'
+
+  useEffect(() => {
```
**issue**: Consider adding a check to ensure `window` is defined before using it to avoid potential errors in server-side rendering.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HeroBottom/HeroBottom.tsx:
```diff
@@ -0,0 +1,309 @@
+  }
+
+  return (
+    <div className={styles.bottomSection}>
```
**praise**: The layout structure is well-organized, making it easy to understand the component hierarchy.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HunterHero.module.scss:
```diff
@@ -1,2 +1,104 @@
+@font-face {
```
**praise**: Including a custom font face adds a unique touch to the design.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/HunterHero.tsx:
```diff
@@ -1,7 +1,27 @@
-export const HunterHero = (props: Props) => {
-  return <div className={styles.wrapper}> HunterHero</div>
+export const HunterHero = async (props: Props) => {
+  const t = await getTranslations(I18nNamespaces.HunterHero)
```
**suggestion**: Consider using a fallback mechanism for translations in case the `getTranslations` function fails.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/TextareaHero/TextareaHero.tsx:
```diff
@@ -0,0 +1,108 @@
+    }, ANIMATION_DURATION);
+
+    return () => clearInterval(interval);
+  }, [currentTag, getRandomTag, onTagChange, selectedTag, mockText, adjustHeight]);
```
**issue**: The useEffect dependency array should include all variables used within the effect to prevent stale closures.

NerdbordMentor commented on apps/platform-app/src/components/hunter-landing/HunterHero/TextareaHero/TextareaHero.tsx:
```diff
@@ -0,0 +1,108 @@
+  };
+
+  return (
+    <div className={styles.textareaWrapper}>
```
**praise**: The placeholder logic is well-implemented, providing a smooth user experience.

NerdbordMentor reviewed (approved):
## Overall Feedback

The code changes effectively implement the required UI for the hero section, both for desktop and mobile views. The text animations in the text area are implemented as specified in the task requirements, using dynamic text animations that align with the provided prototype. The solution includes comprehensive updates across multiple components, ensuring responsiveness and proper integration of new features.

### Things Missing: 

- [] None. All task requirements are met.

### Things That Went Well:

- The implementation of the `TextareaHero` component with dynamic text animations is well-executed and aligns with the prototype.
- The `HeroBottom` component is effectively integrated, providing a responsive tag selection interface with animations.
- The use of SCSS modules for styling ensures maintainability and scalability of the styles.
- The addition of new translations in both English and Polish enhances the application's accessibility and usability.
- The code is well-structured and follows best practices for React and TypeScript development.

Sumick reviewed (approved):
Looks very good, the only thing worth improving here is that we should split HeroSection into smaller components. Currently it's a one huge block of code which will be difficult to maintain later. Let's just improve it before merging.

transcendence12 commented:
This PR refactors the HunterHero components to improve code quality, maintainability, and fix several animation and rendering issues:
Extracted reusable UI components (TagButton, TagsRow, TechIcon, Tooltip) to improve code organization and reusability
Created custom hooks (useTagsAnimation, useTextareaAnimation, useResponsive) to separate logic from presentation
Fixed tooltip positioning issues when switching between mobile and desktop views
Added proper cleanup for DOM elements and timers to prevent memory leaks
Implemented error handling and null checks to make the code more robust
Optimized rendering with useMemo and memo to prevent unnecessary re-renders
Fixed infinite scroll functionality for tag rows on mobile
Improved animation stability for tag cycling in the textarea component
